src/compile/mark-text.ts

```diff
@@ -15,7 +15,10 @@ export namespace text {
       y: { value: 0 },
       width: { field: { group: 'width' } },
       height: { field: { group: 'height' } },
-      fill: { scale: model.scaleName(COLOR), field: model.field(COLOR) }
+      fill: {
+        scale: model.scaleName(COLOR),
+        field: model.field(COLOR, model.fieldDef(COLOR).type === ORDINAL ? {prefn: 'rank_'} : {})
```

Comment:
i'm glad you asked. For bin, I actually generate 3 scales. try the examples to see it. We don't generate a rank.